src/Datadog.Trace.Ci.Shared/GitInfo.cs

```diff
@@ -439,44 +439,62 @@ public static bool TryGetFromObjectFile(string filePath, out GitCommitObject com
-                if (File.Exists(packFile))
+                try
                 {
-                    // packfile format explanation:
-                    // https://codewords.recurse.com/issues/three/unpacking-git-packfiles#:~:text=idx%20file%20contains%20the%20index,pack%20file.&text=Objects%20in%20a%20packfile%20can,of%20storing%20the%20whole%20object.
-
-                    using (var fs = new FileStream(packFile, FileMode.Open, FileAccess.Read, FileShare.Read))
-                    using (var br = new BigEndianBinaryReader(fs))
+                    string packFile = Path.ChangeExtension(packageOffset.FilePath, ".pack");
+                    if (File.Exists(packFile))
                     {
-                        // Move to the offset of the object
-                        fs.Seek(packageOffset.Offset, SeekOrigin.Begin);
-                        int objectSize;
-                        byte[] packData = br.ReadBytes(2);
+                        // packfile format explanation:
+                        // https://codewords.recurse.com/issues/three/unpacking-git-packfiles#:~:text=idx%20file%20contains%20the%20index,pack%20file.&text=Objects%20in%20a%20packfile%20can,of%20storing%20the%20whole%20object.
 
-                        if (packData[0] < 128)
+                        using (var fs = new FileStream(packFile, FileMode.Open, FileAccess.Read, FileShare.Read))
+                        using (var br = new BigEndianBinaryReader(fs))
                         {
-                            objectSize = (int)(packData[0] & 0x0f);
-                            packData = br.ReadBytes(objectSize);
-                        }
-                        else
-                        {
-                            objectSize = (((ushort)(packData[1] & 0x7f)) * 16) + ((ushort)(packData[0] & 0x0f));
-                            packData = br.ReadBytes(objectSize * 100);
-                        }
+                            // Move to the offset of the object
+                            fs.Seek(packageOffset.Offset, SeekOrigin.Begin);
+                            byte[] packData = br.ReadBytes(2);
 
-                        using (var ms = new MemoryStream(packData, 2, packData.Length - 2))
-                        using (var defStream = new DeflateStream(ms, CompressionMode.Decompress))
-                        {
-                            byte[] buffer = new byte[8192];
-                            int readBytes = defStream.Read(buffer, 0, buffer.Length);
-                            defStream.Close();
-                            string strContent = Encoding.UTF8.GetString(buffer, 0, readBytes);
-                            commitObject = new GitCommitObject(strContent);
-                            return true;
+                            // Extract the object size
+                            int objectSize = (int)(packData[0] & 0x0F);
+
+                            if (packData[0] > 128)
```

Comment:
In the previous implementation, this block was executed in the else clause for `packData[0] < 128`, which basically meant `packData[0] >= 128`. Is the new implementation where it's `packData[0] > 128` intended?

Reply:
Oh you're right, good catch!
The comparison is because in the specification from 8 bits, they use 7 for the value and 1 to indicate you should read another byte to continue loading the value. And yes: `128 = 0b1000_0000` so the value is 0, but we should read the next byte.

Reply:
done in afe9ff7